src/me/drton/jmavsim/UDPMavLinkPort.java

```diff
@@ -78,21 +78,25 @@ private static InetAddress getMyHostIPAddress() throws UnknownHostException, Soc
         // Look over all the network interfaces for the appropriate interface.
         Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
         while (networkInterfaces.hasMoreElements()) {
-           NetworkInterface iface = networkInterfaces.nextElement();
-           Enumeration<InetAddress> addresses = iface.getInetAddresses();
-           while (addresses.hasMoreElements()) {
-               InetAddress address = addresses.nextElement();
-               if (!address.isLoopbackAddress()) {
-                   if (address.isSiteLocalAddress()) {
-                       // probably a good one!
-                       return address;
-                   } else {
-                       // Found a non-loopback address that isn't site local.
-                       // Might be link local (private network), but probably better than nothing.
-                       possibleAddress = address;
-                   }
-               }
-           }
+            NetworkInterface iface = networkInterfaces.nextElement();
+            Enumeration<InetAddress> addresses = iface.getInetAddresses();
+            while (addresses.hasMoreElements()) {
+                InetAddress address = addresses.nextElement();
+                if (!address.isLoopbackAddress()) {
+                    if (address.isSiteLocalAddress()) {
+                        // probably a good one!
+                        return address;
+                    } else {
+                        // Found a non-loopback address that isn't site local.
+                        // Might be link local (private network), but probably better than nothing.
+                        possibleAddress = address;
+                        if(address instanceof Inet4Address) {
+                            possibleAddress = address;
+                            break;
```

Comment:
Yes it may fail silently, that seems worth changing. This could also return from the loop as soon as I find an Inet4Address (instead of breaking and continuing to look). @LorenzMeier you would like the user to be able to choose IPv6? This code was added because an IPv6 address got chosen, which made the whole thing crash. I actually haven't been able to reproduce that on my system, so I'm not sure what the issue was, but I can look into that if you'd like.